### PR TITLE
Use HashiCorp Actions Go Build for Consul K8s CLI

### PR DESCRIPTION
**Pull request by t-eckert**
There is a bug in the GH Actions for running acceptance tests. The Go binary that gets set up is not the same version of Go that gets used in subsequent calls to `go`. I have a feeling this is because the downloaded one is lower on the PATH. You can see this issue occurring here: https://github.com/hashicorp/consul-k8s/actions/runs/3331265834/jobs/5510872576

Changes proposed in this PR:
- Use the HashiCorp Go Build Action to build the Consul K8s CLI

How I've tested this PR:
- I will watch it run in GH actions. I had to make a modification to tell it to not run from `main`.

How I expect reviewers to test this PR:
- You can watch it with me.

Checklist:
- [ ] Tests added
- [ ] CHANGELOG entry added 
  > HashiCorp engineers only, community PRs should not add a changelog entry.
  > Entries should use present tense (e.g. Add support for...)

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -344,7 +344,7 @@ jobs:
 
   acceptance:
     needs: [ get-product-version, dev-upload-docker, get-go-version ]
-    uses: hashicorp/consul-k8s/.github/workflows/reusable-acceptance.yml@main
+    uses: hashicorp/consul-k8s/.github/workflows/reusable-acceptance.yml@te/use-hc-actions-build
```
Comment (t-eckert) [inline .github/workflows/test.yml:347]: This will change before I merge.

**Comment by t-eckert** [comment]
@wilkermichael will we not be running acceptance tests for PRs in GH actions for now? If so, I'll close this.

**Comment by wilkermichael** [comment]
We'll still need this, this reusable acceptance test will be used in the nightlies as well

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -344,7 +344,7 @@ jobs:
 
   acceptance:
     needs: [ get-product-version, dev-upload-docker, get-go-version ]
-    uses: hashicorp/consul-k8s/.github/workflows/reusable-acceptance.yml@main
+    uses: hashicorp/consul-k8s/.github/workflows/reusable-acceptance.yml@536f034d6647fdadca8892036d0b06e28d1c9a57
```
Comment (t-eckert) [inline .github/workflows/test.yml:347]: Change before merging.

**Comment by t-eckert** [comment]
Looks like we will not be doing acceptance tests on PRs with GH actions. Closing this PR.

**Pull request closed without merging**